pubsub_features/environment.py

```diff
@@ -0,0 +1,83 @@
+import json
+import logging
+import time
+import uuid
+from datetime import datetime
+
+import requests
+
+from config import Config
+from utilties.rabbit_context import RabbitContext
+
+
+def before_all(_):
+    logging.getLogger('pika').setLevel('ERROR')
+    logging.getLogger('paramiko').setLevel('ERROR')
+    logging.captureWarnings(True)
+
+
+def before_scenario(context, scenario):
+    if "skip" in scenario.effective_tags:
+        scenario.skip("Marked with @skip")
+        return
+
+    context.test_start_local_datetime = datetime.now()
+    context.action_plan_id = str(uuid.uuid4())
+    _clear_down_all_queues()
+
+
+def get_msg_count(queue_name):
+    uri = f'http://{Config.RABBITMQ_HOST}:{Config.RABBITMQ_MAN_PORT}/api/queues/%2f/{queue_name}'
+    response = requests.get(uri, auth=(Config.RABBITMQ_USER, Config.RABBITMQ_PASSWORD))
+    response.raise_for_status()
+    response_data = json.loads(response.content)
+
+    return response_data['messages']
+
+
+def _clear_down_all_queues():
+    all_queues = _get_all_queues()
+
+    for queue in all_queues:
+        # keep killing this Delayed queue, just to stop it redelivering anything in some mad race condition
+        _clear_down_queue(Config.RABBITMQ_DELAYED_REDELIVERY_QUEUE)
+        _clear_down_queue(queue)
+
+
+def _get_all_queues():
+    uri = f'http://{Config.RABBITMQ_HOST}:{Config.RABBITMQ_MAN_PORT}/api/queues/%2f/'
+    response = requests.get(uri, auth=(Config.RABBITMQ_USER, Config.RABBITMQ_PASSWORD))
+    response.raise_for_status()
+    response_data = json.loads(response.content)
+
+    return [queue['name'] for queue in response_data]
+
+
+def _clear_down_queue(queue_name):
```

Comment:
i think @AdamHawtin @ryangrundy7 changed the way we clear rabbit queues as it was timing out when trying to do this for large amount (millions) of messages... https://trello.com/c/BTBz1ALz/590-completely-purge-rabbit-after-performance-test-runs

Reply:
I think this functions is unused anyway

Reply:
just had a conversation about this maybe something we can sole at a later point, but is good for here